### ISSUE TITLE
Zone issue #537 using Electron menus resurfacing in Angular v7.2.8

### ISSUE THREAD

**Issue by pamtbaau** [angular/zone.js#1200]
It seems like issue [#537](https://github.com/angular/zone.js/issues/537) is resurfacing after upgrading to angular v7.2.8.

Angular does not detect changes anymore when change occurs after selecting Electron menu.

**Comment by JiaLiPassion** [comment]
@pamtbaau, did you load `dist/zone-patch-electron`? And could you post a sample reproduce repo?

**Comment by pamtbaau** [comment]
Yes, import `zone.js/dist/zone-patch-electron`  is still in my polyfill. I'm now stepping back in my package upgrades to see where the issue started.

**Comment by pamtbaau** [comment]
@JiaLiPassion It seems to be caused by a change in Electron 4+.

Working       : Angular:7.2.8, Zone:0.8.29, Electron:3.1.6
Not working : Angular:7.2.8, Zone:0.8.29, Electron:4.0.0
Not working : Angular:7.2.8, Zone:0.8.29, Electron:4.0.8
Working       : Angular:7.2.8, Zone:0.8.29, Electron:3.1.6
Working       : Angular:7.2.8, Zone:0.8.29, Electron:3.1.0

**Comment by JiaLiPassion** [comment]
@pamtbaau, thanks, I will check it.

**Comment by pamtbaau** [comment]
@JiaLiPassion Any progress you might be able to share?

**Comment by JiaLiPassion** [comment]
@pamtbaau, sorry I haven't checked this one yet, could you post a reproduce repo for me? Thanks!

**Comment by pamtbaau** [comment]
Here you go: [https://github.com/pamtbaau/broken-zone](https://github.com/pamtbaau/broken-zone)

**Comment by JiaLiPassion** [comment]
@pamtbaau, thanks, I will test it.

**Comment by pamtbaau** [comment]
@JiaLiPassion Any insights you can share?

**Comment by JiaLiPassion** [comment]
@pamtbaau, sorry for late reply, I am working on the other issues, I tried your repo, there is one dependency `xelectron` is not found in `npm registry`, is this some special package?

**Comment by pamtbaau** [comment]
Yes, that is a special package with a typo... Don't know how that electron package got there. Fixed the repo.

**Comment by JiaLiPassion** [comment]
@pamtbaau, thanks, I will try to fix this one.

**Comment by JiaLiPassion** [comment]
@pamtbaau, you can try the attached zip file, override your `node_modules/zone.js/dist` folder.
[zonejs-dist.zip](https://github.com/angular/zone.js/files/3059370/zonejs-dist.zip)

**Comment by pamtbaau** [comment]
Fix works in 2 of the 3 electron apps... The two are very simple, but the 3rd is quite complex. I'll have to do some homework...

**Comment by JiaLiPassion** [comment]
@pamtbaau, thanks for the confirmation, I only patched `renderer.on` method, if some cases not working, I may need to patch more methods.

**Comment by pamtbaau** [comment]
@JiaLiPassion Do you recall the patch you created for issue [#537](https://github.com/angular/zone.js/issues/537)? That patch worked perfectly on all my apps. 

Did you perhaps patch more methods in that patch?

**Comment by JiaLiPassion** [comment]
@pamtbaau, yes, in https://github.com/angular/zone.js/issues/537, the `patch` patched the `CallbackRegistry` which is a centralised location I can patch, but from `Electron 4`, this `CallbackRegistry` can not be accessed from outside, so I have to patch renderer, I can patch more methods, that is not difficult, if you can give me more information about in which case still not work, it will be great help.

**Comment by pamtbaau** [comment]
After dissenting the failing app for hours... I finally found a reproducible case where Angular's zone remains broken... I have update my repo.

When importing a class with a static property calling Electrons `remote`,  change detection after a popup will fail. 

Try to comment/uncomment line 5 in settings.ts to see zone breaking..

**Comment by pamtbaau** [comment]
@JiaLiPassion Any news?

**Comment by biandtlp** [comment]
Same problem here

**Comment by JiaLiPassion** [comment]
I will look this issues later this week, sorry for the delay.

**Comment by pamtbaau** [comment]
@JiaLiPassion Any news?

PS. If you feel I'm becoming a pita you may say so...

**Comment by JiaLiPassion** [comment]
@pamtbaau, sorry I am working on some other issues recently(moving zone to angular repo), and all other pending issues need to wait for that unless some very urgent bugs, sorry for the delay, I will check this issue as soon as I finish the working PR.

**Comment by pamtbaau** [comment]
@JiaLiPassion Any target date for the working PR you can share?

**Comment by JiaLiPassion** [comment]
It should be in two weeks. If this issue is very urgent, I can make a patch branch for you.

**Comment by pamtbaau** [comment]
@JiaLiPassion Thanks for the offer, but there is no need to increase your workload just for my project... I can stay on a earlier version of Electron. Having a rough estimate I can leave you in peace...

**Comment by JiaLiPassion** [comment]
@pamtbaau, sorry for the very late reply, I checked your updated source, the reason why it is not working is, the code here
```
  private static breaksZone = remote.app.getPath('userData');
```

is loaded before `ngZone` created, and when `electron` load `remote module` it will create the `ipcRenderer.on` event handler outside of `angular zone`, and there is no elegant way to change this behavior, so you have to make sure that `electron` load those stuff inside of `ngZone`.
